packages/main/src/plugin/troubleshooting.ts

```diff
@@ -0,0 +1,95 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import AdmZip from 'adm-zip';
+import moment from 'moment';
+import * as os from 'node:os';
+import * as fs from 'node:fs';
+
+const SYSTEM_FILENAME = 'system';
+
+export interface FileMap {
+  fileName: string;
+  content: string;
+}
+
+export type LogType = 'log' | 'warn' | 'trace' | 'debug' | 'error';
+
+export class Troubleshooting {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  constructor(private apiSender: any) {}
+
+  async saveLogsToZip(fileMaps: FileMap[], destination: string): Promise<void> {
+    if (fileMaps.length === 0) return;
+
+    const zip = new AdmZip();
+    fileMaps.forEach(fileMap => {
+      zip.addFile(`${fileMap.fileName}.txt`, Buffer.from(fileMap.content, 'utf8'));
+    });
+    zip.writeZip(destination);
+    console.log(`Zip file saved to ${destination}`);
+  }
+
+  getConsoleLogs(consoleLogs: { logType: LogType; message: string }[]): FileMap[] {
+    const content = consoleLogs.map(log => `${log.logType} : ${log.message}`).join('\n');
+    return [{ fileName: generateLogFileName('console'), content }];
+  }
+
+  async getSystemLogs(): Promise<FileMap[]> {
+    switch (os.platform()) {
+      case 'darwin':
+        return this.getLogsFromFiles(
+          ['launchd-stdout', 'launchd-stderr'],
+          `${os.homedir()}/Library/Logs/Podman Desktop`,
+        );
+      case 'win32':
+        return this.getLogsFromFiles(['podman-desktop'], `${os.homedir()}/AppData/Roaming/Podman Desktop/logs`);
+      default:
+        console.log('Unsupported platform for retrieving system logs');
+        return [];
+    }
+  }
+
+  private async getFileSystemContent(filePath: string, logName: string): Promise<FileMap> {
+    try {
+      const content = await fs.promises.readFile(filePath, 'utf8');
```

Comment:
I am getting an `Error: ENOENT: no such file or directory, open 'C:\Users\{pc}\AppData\Roaming\Podman Desktop\logs\podman-desktop.log'` 

Maybe we should check if the file exist before trying to open it ?


> I am on windows 11

Reply:
Added a simple check to see if the file exists, if it doesnt, it'll skip it and output to log we were unable to find it.